micromanager_gui/main_window.py

```diff
@@ -245,46 +245,58 @@ def load_cfg(self):
 
     def _refresh_camera_options(self):
         cam_device = self._mmc.getCameraDevice()
+        if not cam_device:
+            return
         cam_props = self._mmc.getDevicePropertyNames(cam_device)
         if "Binning" in cam_props:
-            self.bin_comboBox.clear()
             bin_opts = self._mmc.getAllowedPropertyValues(cam_device, "Binning")
-            self.bin_comboBox.addItems(bin_opts)
-            self.bin_comboBox.setCurrentText(
-                self._mmc.getProperty(cam_device, "Binning")
-            )
+            with blockSignals(self.bin_comboBox):
+                self.bin_comboBox.clear()
+                self.bin_comboBox.addItems(bin_opts)
+                self.bin_comboBox.setCurrentText(
+                    self._mmc.getProperty(cam_device, "Binning")
+                )
 
         if "PixelType" in cam_props:
-            self.bit_comboBox.clear()
             px_t = self._mmc.getAllowedPropertyValues(cam_device, "PixelType")
-            self.bit_comboBox.addItems(px_t)
-            if "16" in px_t:
-                self.bit_comboBox.setCurrentText("16bit")
-                self._mmc.setProperty(cam_device, "PixelType", "16bit")
+            with blockSignals(self.bit_comboBox):
+                self.bit_comboBox.clear()
+                self.bit_comboBox.addItems(px_t)
+                self.bit_comboBox.setCurrentText(
+                    self._mmc.getProperty(cam_device, "PixelType")
+                )
 
     def _refresh_objective_options(self):
         if "Objective" in self._mmc.getLoadedDevices():
-            self.objective_comboBox.clear()
-            self.objective_comboBox.addItems(self._mmc.getStateLabels("Objective"))
+            with blockSignals(self.objective_comboBox):
+                self.objective_comboBox.clear()
+                self.objective_comboBox.addItems(self._mmc.getStateLabels("Objective"))
+                self.objective_comboBox.setCurrentText(
+                    self._mmc.getProperty("Objective", "Label")
+                )
 
     def _refresh_channel_list(self, channel_group: str = None):
         if channel_group is None:
             channel_group = self._get_channel_group()
         if channel_group:
             channel_list = list(self._mmc.getAvailableConfigs(channel_group))
-            self.snap_channel_comboBox.addItems(channel_list)
-
-    def _on_system_configuration_loaded(self):
-        self._refresh_camera_options()
-        self._refresh_objective_options()
-        self._refresh_channel_list()
-        self._refresh_positions()
+            with blockSignals(self.snap_channel_comboBox):
+                self.snap_channel_comboBox.addItems(channel_list)
+                self.snap_channel_comboBox.setCurrentText(
+                    self._mmc.getCurrentConfig("Channel")
+                )
```

Comment:
Could also use `getCurrentConfigFromCache` for faster method, but this should be closer to the ground truth and since refresh should only be called rarely I think it's best to do this.